### DISCUSSION TITLE
Is this project still live?

### DISCUSSION THREAD

**Discussion by WiggyB** [greghope667/square-off-remote#9] [General]
Hey Guys,

I been looking to get around the square off app myself and come across your project.
Does this code still work for you? I have the "Swap C80" board. It connects, and the board beeps, but that's it. 
It's give a "4#4095.00*" as the battery level (When plugged in).
And it claims to be making the 4 random moves, but the board isn't doing anything.
I've been talking to another dev that has messed with square off boards, and it looks as if encryption has been added to the communication. 
Does that track with what you have seen?

**Comment by greghope667** [comment]
Hi,

This project has been on hold for a bit as I've been distracted by other (mostly work related) things. The code currently still works for me on my board, but I'm using the older "Grand Kingdom" board with (presumably) older firmware.

That battery response is differently formatted to what I get which suggests the communication protocols have changed for newer boards. The fact that it connects and gives any response at all is a good sign - it suggests that the newer protocol may be similar and share some commands. Maybe just the move command syntax has changed? If the communication were fully encrypted then it wouldn't be possible to connect/get any response. (I guess they could have encrypted/locked out just some parts if they disliked people like me hacking on their hardware.)

I see the "4 random moves" mode doesn't work for you. Have you tried playing against the board as a game? They work slightly differently in how the comms are handled ('Live' mode/'Game' mode).

To work out the protocol originally I MITM'd the Bluetooth connection between my phone and the board. If the protocol has actually changed, a similar tactic for the newer boards might work for someone with the right equipment.
